### PR TITLE
[receiver/redisreceiver] Add redis.version as resource attribute

### PR DESCRIPTION
**Pull request by singku**
**Descriptions**
- Add redis.version resource attribute.

**Related Discussions**
#11090

**Documentation**
In-place documents added. There are also auto generated document for updated metrics.

### CODE REVIEW

File: receiver/redisreceiver/metadata.yaml [hunk 1]
```diff
@@ -15,14 +20,14 @@ attributes:
 
 metrics:
   redis.maxmemory:
-    enabled: false
+    enabled: true
```
Comment (dmitryax) [inline receiver/redisreceiver/metadata.yaml:23]: Why changing them to enabled by default? I don't remember we agreed on this, and I still don't think they are essential metrics

Comment (singku) [reply receiver/redisreceiver/metadata.yaml:23]: I thought adding resource attribute is already a breaking change, so why not change them to true by default as well.

Comment (dmitryax) [reply receiver/redisreceiver/metadata.yaml:23]: Because I don't believe that these metrics are important enough to be emitted by default

Comment (singku) [reply receiver/redisreceiver/metadata.yaml]: I changed them back

File: receiver/redisreceiver/redis_scraper.go [hunk 2]
```diff
@@ -142,6 +143,15 @@ func (rs *redisScraper) recordKeyspaceMetrics(ts pcommon.Timestamp, inf info) {
 	}
 }
 
+// getRedisVersion retrieves version string from 'redis_version' Redis info key-value pairs
+// e.g. "redis_version:5.0.7"
+func (rs *redisScraper) getRedisVersion(inf info) string {
+	if str, ok := inf["redis_version"]; ok {
+		return str
+	}
+	return "unknown_version"
```
Comment (dmitryax) [inline receiver/redisreceiver/redis_scraper.go:152]: Maybe just `unknown`?

File: receiver/redisreceiver/redis_scraper.go [hunk 3]
```diff
@@ -90,7 +90,8 @@ func (rs *redisScraper) Scrape(context.Context) (pmetric.Metrics, error) {
 	rs.recordKeyspaceMetrics(now, inf)
 	rs.recordRoleMetrics(now, inf)
 	rs.recordCmdStatsMetrics(now, inf)
-	return rs.mb.Emit(), nil
+	resourceOpts := []metadata.ResourceMetricsOption{metadata.WithRedisVersion(rs.getRedisVersion(inf))}
+	return rs.mb.Emit(resourceOpts...), nil
```
Comment (dmitryax) [inline receiver/redisreceiver/redis_scraper.go:94]:
```suggestion
	return rs.mb.Emit(metadata.WithRedisVersion(rs.getRedisVersion(inf))), nil
```

**Review by dmitryax — APPROVED** [review]

File: unreleased/redisreceiver-version.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (codeboten) [inline unreleased/redisreceiver-version.yaml:2]: Is this really a breaking change? I would have expected this to be an enhancement as it's additive.

Comment (dmitryax) [reply unreleased/redisreceiver-version.yaml:2]: It depends on how we define breaking changes for emitted data. I would say it is a breaking change. It introduces new attribute which can be unexpected on the metric backends and generated new MTS's

Comment (singku) [reply unreleased/redisreceiver-version.yaml:2]: So as @dmitryax described, this attributed will be addd to all emitted metrics as an additional attribute/label. If consumer has strict attributes check, their workflows will be broken.

Comment (dmitryax) [reply unreleased/redisreceiver-version.yaml:2]: I'm ok with moving it to enhancement given that it's not a significantly breaking for the emitted data

Comment (codeboten) [reply unreleased/redisreceiver-version.yaml:2]: i would expect a changing an existing attribute to be breaking, but not adding a new one. 

> Adding of new metrics, spans, span events or log records and adding of new attributes to spans, span events, log records or resources are considered additive, non-breaking changes and are always allowed for Unstable and Stable instrumentations.

https://github.com/open-telemetry/opentelemetry-specification/blob/cba6938951a789f861303d1b4ef3634ef00b9aa5/specification/telemetry-stability.md

Comment (dmitryax) [reply unreleased/redisreceiver-version.yaml:2]: Ok let's move it to enhancement

Comment (singku) [reply unreleased/redisreceiver-version.yaml]: Updated.

**Review by codeboten — APPROVED** [review]

**Pull request merged**